src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -962,11 +962,41 @@ public override BoundNode VisitPassByCopy(BoundPassByCopy node)
         public override BoundNode VisitIsPatternExpression(BoundIsPatternExpression node)
         {
             Debug.Assert(!IsConditionalState);
-            VisitRvalue(node.Expression);
 
             bool negated = node.Pattern.IsNegated(out var pattern);
             Debug.Assert(negated == node.IsNegated);
 
+            if (VisitPossibleConditionalAccess(node.Expression, out var stateWhenNotNull))
+            {
```

Comment:
I assume that we can assert that the state is not conditional. Consider adding assertion

Reply:
Done